### ISSUE TITLE
Convert panic on TerminalAction::OpenFile to recoverable error

### ISSUE THREAD

**Issue by ReeceMcMillin** [zellij-org/zellij#973]
Currently, `zellij_server::os_input_output::spawn_terminal` panics if certain environment variables aren't set.

```rust
if env::var("EDITOR").is_err() && env::var("VISUAL").is_err() {
    panic!("Can't edit files if an editor is not defined. To fix: define the EDITOR or VISUAL environment variables with the path to your editor (eg. /usr/bin/vim)");
}
```

The error message guides the user to the right solution, but this feels like an opportunity to handle the error gracefully. In a situation like this, I might expect to receive a prompt to set those variables for the session. That would require #708 so may not be immediately available.

As a quick hack, I removed the panic in favor of calling `less`. This isn't exactly the goal of the plugin, but gets a lot closer to behaving that way when the environment variables aren't set. The warning currently shown by the panic should still be displayed, but I'm not familiar enough with the architecture to get that up and running yet.

```rust
let command = if env::var("EDITOR").is_err() && env::var("VISUAL").is_err() {
    PathBuf::from("less")
} else {
    PathBuf::from(env::var("EDITOR").unwrap_or_else(|_| env::var("VISUAL").unwrap()))
};
```

Overall, I think this is a good move as the error can be predicted and handled within the session which doesn't need to rise to the level of a panic. Happy to dig in and explore options, but I'd be a first-time contributor and have been told this might not be well suited for a first issue.

**Comment by imsnif** [comment]
Hey - I actually just hit this myself after not thinking about this code path for a long time (formatted my machine and forgot to set the variable), and I agree this is not the best experience.

The issue here is that as you say, we don't yet have a system in place to display errors to the user in app. So while this error is predictable and recoverable, we still need a way to communicate it.

Your solution with less is great (assuming less itself is installed on the machine ofc), but as you say we still need to display it somehow. Maybe something like `echo "<MESSAGE>" | less`? (checking before that both `echo` and `less` exist on the machine? A bit hacky, but definitely better than what we have now. Wdyt?

**Comment by ReeceMcMillin** [comment]
@imsnif yeah, I think that'd be great! Just riffing on that idea, we can follow that with a check for an `editor` binary along these lines with pretty high compatibility with the [Linux standard](https://refspecs.linuxfoundation.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/command.html) (using `read` and `more`). I also just formatted my machine and the fresh install had an editor _binary_ but no variable set. It's not a perfect promise, but as far as holdovers go it seems like a pretty reasonable compromise. A tiny shell example where `$1` is a filename might be something like
```bash
#!/bin/bash
command -v editor $1 > /dev/null 2>&1 \
     || read -p "'editor' binary not found, opening $1 with 'more'. [Enter]" RESP &&
        more $1
```
I haven't had a chance to throw this into Zellij yet so there might be issues I haven't considered, but I think that's about as comfortable a compatibility promise as we're going to get without overdoing it. I can hack something together if you think that's a worthwhile direction, but I'm totally open to exploring other options if you have reservations. Thanks!

**Comment by imsnif** [comment]
Sure, sounds good. But I'd rather not have all this logic in bash... can we check the env variables and the presence of executables with Rust and then do the right thing?

**Comment by ReeceMcMillin** [comment]
Totally, that's the plan! The bash was just a quick example of the idea. I'll play with this and update later, thanks!
